server/controllers/email-report/weeklySummaryReport.js

```diff
@@ -0,0 +1,217 @@
+*/
+function getRegisteredPatientsNumber() {
+
+  var period = new Period(new Date());
+
+  var week = period.periods.week.limit;
+
+  var monday = week.start();
+  var sunday = week.end();
+
+  var Lastweek = period.periods.lastWeek.limit;
+
+  var lastmonday = Lastweek.start();
+  var lastsunday = Lastweek.end();
+
+  //patients registered this week
+  var sql = `SELECT
+              subquery1.NumberOfpatientsThisWeek, subquery2.NumberOfpatientsLastWeek
+            FROM
+              (
+                SELECT count(uuid) as NumberOfpatientsThisWeek
+                FROM patient
+                WHERE (registration_date BETWEEN ? AND ? )
+              ) as subquery1,
+              (
+                SELECT count(uuid) as NumberOfpatientsLastWeek
+                FROM patient
+                WHERE (registration_date BETWEEN ? AND ? )
+              ) as subquery2`;
+
+  var values = [monday, sunday, lastmonday, lastsunday];
+
+  var thisWeekPatients = [];
+
+  return db.exec(sql, values);
+
+}
+
+//first and last invoice per day for a week
+function MaxMinInvoiceDate() {
+
+  var sql = `
+    SELECT  MIN(i.date) as minDate, MAX(i.date) as maxDate
+    From invoice i
+    WHERE DATE(i.date) BETWEEN ? AND ?
+    GROUP BY DATE(i.date)
+  `;
+
+  var period = new Period(new Date());
+  var week = period.periods.week.limit;
+
+  var monday = week.start();
+  var sunday = week.end();
+
+  return db.exec(sql, [monday, sunday]);
+}
+
+//test to see how to the report looks like
+function view1(req, res, next) {
+
+  const options = {
+    renderer: 'pdf',
+    saveReport: false,
+    filename: 'test_summary_report',
+    orientation: 'portrait',
+  };
+
+  //patient registered
+  getRegisteredPatientsNumber().then(_patientsData => {
+
+    //services incomes
+    loadServicesIncome().then(_servicesIncome => {
+
+      //max and min date of invoice each day for a week
+      MaxMinInvoiceDate().then(_MaxMinInvoiceDate => {
+
+        //rendering the report
+        var report = new ReportManager(TEMPLATE, req.session, options);
+
+        if (_patientsData.lenght === 0) {
+          _patientsData = { "NumberOfpatientsThisWeek": 0, "NumberOfpatientsLastWeek": 0 };
+        } else {
+          _patientsData = _patientsData[0];
+        }
+
+        var data = {
+          patientsData: _patientsData,
+          servicesIncome: _servicesIncome,
+          MaxMinInvoiceDate: _MaxMinInvoiceDate
+        }
+
+        report.render(data).then((result) => {
+          res.set(result.headers).send(result.report);
+        });
+
+      });
+    });
+
+  })
+  .catch(next)
+  .done();
+
+
+}
+
+
+//one of the reports that bhima send by email
+function weeklySummaryReport(currentSession) {
+
+  const options = {
+    renderer: 'pdf',
+    saveReport: false,
+    filename: 'test_summary_report.pdf',
+    orientation: 'portrait',
+  };
+
+  //patient registered
+  return getRegisteredPatientsNumber().then( (_patientsData) => {
```

Comment:
This should be a promise chain to avoid making a giant tree:
```js
const data = {};
return getRegisteredPatientsNumber()
  .then(patientsData => {
     data.patientsData = patientsData;
     return loadServicesIncome();
  .then(servicesIncome => {
     data.servicesIncome = servicesIncome;
     return MaxMinInvoiceDate();
  })
  .then(maxMinInvoiceDate => {
    data.maxMinInvoiceDate = maxMinInvoiceDate;
    return report.render(data);
   });
```